screens/DetailsScreen.js

```diff
@@ -1,13 +1,79 @@
 import React from 'react'
 import { StyleSheet, Text, View } from 'react-native'
+import MapView from 'react-native-maps'
 
 export class DetailsScreen extends React.Component {
 
   constructor (props) {
     super(props)
     this.state = {
-      date: new Date()
+      date: new Date(),
```

Comment:
I'm not really into this approach. IMHO it would be betted to sth like:

```
state = { loaded: false }
```
and then 
```
fetch("...").then((response) => response.json()).then(r => {
	this.setState({ loaded: true });
	this.response = r;
}) 
```

And then in render
```
{this.state.loaded ? <View>sth</View> : <ActivityIndicator/>}
```
Finally mind that there's no need to persist all fetched in state. `loaded` flag is enough

Reply:
And last thing: I do not consider really required to map data from JSON to other format. I suppose it's fully OK to pass it to view directly